be/src/olap/compaction.cpp

```diff
@@ -136,7 +136,12 @@ Status Compaction::do_compaction_impl(int64_t permits) {
     int64_t current_max_version;
     {
         std::shared_lock rdlock(_tablet->get_header_lock());
-        current_max_version = _tablet->rowset_with_max_version()->end_version();
```

Comment:
when will rowset_with_max_version == nullptr?

Reply:
This seems to happen in two cases:
1. The tablet has no rowset.
2. The version of max version rowset no longer exists in the tablet.

In fact, I did not carefully analyze why it occurs. Because other `tablet->rowset_with_max_version()` all compare whether the return value is null, so I also compare here :)

Reply:
1. The tablet should not have rowset because when the table is created BE will add 0-1 rowset to the tablet.
2. I think we should be careful with this case... Maybe should deep dive some code.

Reply:
> 1. The tablet should not have rowset because when the table is created BE will add 0-1 rowset to the tablet.
> 2. I think we should be careful with this case... Maybe should deep dive some code.

I agree, these two cases are not normal, maybe the tablet was deleted during compaction, or invalid?

At present this is the previous solution, there may be a better way.